server/athenian/api/models/state/models.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+
+from sqlalchemy import Column, Integer, JSON, String, TIMESTAMP
+from sqlalchemy.ext.declarative import declarative_base
+
+
+# The following two classes compensate the absent ORM layer in databases.Database.
+
+class Refresheable:
+    """Mixin to invoke default() and onupdate() on all the columns."""
+
+    class Context:
+        """Pretend to be a fully-featured SQLAlchemy execution context."""
+
+        def __init__(self, parameters: dict):
+            """init"""
+            self.current_parameters = parameters
+
+    def create_defaults(self):
```

Comment:
Isn't this done automatically? I mean when creating an instance of `RepositorySet`, doesn't the unset fields have the default value? AFAIS sqlalchemy already handles a `default` kwarg. Are you overriding its behavior?

BTW it's not the top that we need to right these behaviors our own :(

Reply:
This indeed sucks. I left a comment about this on line 7.

SQLAlchemy has two layers, and those fancy defaults and onupdate-s belong to the ORM. However, `databases`, the asyncio DB engine we are using, does not support the ORM layer.

Reply:
Ah ok, it's clear now. It was weird to me when I read it, but now I understand that it's specifically for the asyncio DB engine. That would be great if you could specify it 👍